### DISCUSSION TITLE
I'd like to get an example of 'attachMapLayer'.

### DISCUSSION THREAD

**Discussion by Myzik-Dev** [iTwin/itwinjs-core#200] [Q&A] [answered]
I'd like to get an example of 'attachMapLayer'.

I tested it with the following code, but the map is not shown.

```
const testMapLayer:MapLayerProps= {subLayers:[],formatId:"BingMaps",name:"Bing Maps: Aerial Imagery with labels",url:"https://dev.virtualearth.net/REST/v1/Imagery/Metadata/AerialWithLabels?o=json&incl=ImageryProviders&key={TestKey}",transparentBackground:false,isBase:true, visible: true};

style.attachMapLayer(testMapLayer,true);
```

I'd appreciate it if you could tell me the right way to use it.

**Comment by pmconne** [comment]
An example can be found in [MapLayerTool.ts](https://github.com/imodeljs/imodeljs/blob/master/core/frontend-devtools/src/tools/MapLayerTool.ts). Perhaps @RBBentley could supply a simplified example.
If all you want is to enable Bing maps, see [Viewport.changeBackgroundMapProps](https://github.com/imodeljs/imodeljs/blob/11ac2575dd2b505e8f8c6270ede2b156e5970861/core/frontend/src/Viewport.ts#L1416).
Can you check the network tab and console for errors? `key={TestKey}` looks suspicious.

**Comment by RBBentley** [comment, marked as answer]
If you only want to turn on Bing map display you could do something like Viewport.changeBackgroundMapProps({});  I believe the default properties are bing maps with hybrid display.   Or more explicitly you could use Viewport.changeBackgroundMapProps({providerName: "BingProvider", providerData: { mapType: 2 }});  - mapType 2 would be the aerial display mode.

**Reply by pmconne** [reply]
For reference: [BackgroundMapType.Aerial](https://github.com/imodeljs/imodeljs/blob/9adbab72545397f33ddffba0b3853c45d36cc128/core/common/src/BackgroundMapSettings.ts#L16).

**Comment by Myzik-Dev** [comment]
Thank you for your answer. 👍 

It was my mistake.
It was a mistake that commented out the calling function.
